### PR TITLE
Added initial chat delete tests

### PR DESCRIPTION
**Pull request by TheHollidayInn**
These are the tests for deleteChatMessage in #6509.

### CODE REVIEW

File: website/src/controllers/api-v3/chat.js [hunk 1]
```diff
@@ -256,4 +257,60 @@ api.flagChat = {
...
+    let validationErrors = req.validationErrors();
+    if (validationErrors) throw validationErrors;
+
+    let group = await Group.getGroup({user, groupId, fields: 'chat'});
+    if (!group) throw new NotFound(res.t('groupNotFound'));
+
+    let message = _.find(group.chat, {id: chatId});
+    if (!message) throw new NotFound(res.t('messageGroupChatNotFound'));
+
+    if (user._id !== message.uuid && !(user.backer && user.contributor.admin)) {
```
Comment (paglias) [inline website/src/controllers/api-v3/chat.js]: why check for both backer && admin, admin should be more than enough

File: website/src/controllers/api-v3/chat.js [hunk 2]
```diff
@@ -256,4 +257,60 @@ api.flagChat = {
...
+    );
+
+    if (chatUpdated) {
+      group = group.toJSON();
+      _.remove(group.chat, function removeChat (chat) {
+        return chat.id === chatId;
+      });
+      res.json(group.chat);
+    } else {
+      res.send(204);
```
Comment (paglias) [inline website/src/controllers/api-v3/chat.js]: res.respond(200, {})

File: website/src/controllers/api-v3/chat.js (hunk 1 above)
Comment (TheHollidayInn) [inline website/src/controllers/api-v3/chat.js]: Make sense. I wasn't sure if there was a difference, so I kept it from v2.

**Comment by TheHollidayInn** [comment]
Updated!

File: website/src/controllers/api-v3/chat.js [hunk 3]
```diff
@@ -256,4 +257,60 @@ api.flagChat = {
...
+/**
+ * @api {delete} /groups/:groupId/chat/:chatId Delete chat message from a group
+ * @apiVersion 3.0.0
+ * @apiName DeleteChat
+ * @apiGroup Chat
+ *
+ * @apiParam {string} groupId The group _id (or 'party')
+ * @apiParam {string} chatId The chat _id
+ *
+ * @apiSuccess {Array} An empty array
```
Comment (crookedneighbor) [inline website/src/controllers/api-v3/chat.js]: It returns an empty array?

File: test/api/v3/integration/chat/DELETE-chat_id.test.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      expect(messages).is.an('array');
+      expect(messages).to.not.include(nextMessage);
+    });
+
+    it('returns empty when previous message parameter is passed and the last message was deleted', async () => {
+      expect(await user.del(`/groups/${groupWithChat._id}/chat/${nextMessage.id}?previousMsg=${nextMessage.id}`))
+        .to.be.empty;
+    });
+
+    it('returns empty when previous message parameter is passed and the chat is updated', async () => {
```
Comment (crookedneighbor) [inline test/api/v3/integration/chat/DELETE-chat_id.test.js]: returns empty? Looks to me like it returns an array.

File: test/api/v3/integration/chat/DELETE-chat_id.test.js [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+      expect(await user.del(`/groups/${groupWithChat._id}/chat/${nextMessage.id}?previousMsg=${message.id}`))
```
Comment (crookedneighbor) [inline test/api/v3/integration/chat/DELETE-chat_id.test.js]: Didn't we decide to do `await expect` instead of `expect(await`?

**Comment by crookedneighbor** [comment]
Aside from my few comments, looks good.

File: test/api/v3/integration/chat/DELETE-chat_id.test.js (hunk 5 above)
Comment (TheHollidayInn) [inline test/api/v3/integration/chat/DELETE-chat_id.test.js]: Yes, sorry :P

**Comment by TheHollidayInn** [comment]
Fixed. Thanks for looking it over.

File: website/src/controllers/api-v3/chat.js [hunk 6]
```diff
@@ -256,4 +257,61 @@ api.flagChat = {
...
+      {_id: group._id},
+      {$pull: {chat: {id: chatId} } }
+    );
+
+    if (chatUpdated) {
+      group = group.toJSON();
+      _.remove(group.chat, function removeChat (chat) {
+        return chat.id === chatId;
+      });
+      res.json(group.chat);
```
Comment (paglias) [inline website/src/controllers/api-v3/chat.js:310]: both here and at line 312 you should use res.respond :) res.send won't send the response as JSON

Comment (paglias) [inline website/src/controllers/api-v3/chat.js:310]: and we use res.respond in place of res.json because we can manipulate the response or apply defaults

**Pull request merged**